components/auth/SocialAuthGroup.tsx

```diff
@@ -0,0 +1,25 @@
+import { getProviders } from "next-auth/react";
+import React from "react";
+import SocialAuthButton from "@/components/auth/SocialAuthButton";
+import google from "@/public/icons/google.svg";
+import github from "@/public/icons/github.svg";
+export const SocialAuthGroup = async () => {
+  const providers = await getProviders();
+  return (
+    <div>
+      {" "}
+      <SocialAuthButton
+        className="social-login-btn social-login-shadow"
+        icon={google}
+        text={`${providers?.google.name || ""}`}
+        provider={providers?.google.name || ""}
+      ></SocialAuthButton>
+      <SocialAuthButton
+        className="social-login-btn social-login-shadow"
+        icon={github}
+        text={`${providers?.github.name || ""}`}
+        provider={providers?.github.name || ""}
+      />
```

Comment:
If we want to map over our providers- then importing the providers makes sense. But since we'll just be using a couple different ones- let's just make our code simple and clear:

```js
export const SocialAuthGroup = async () => {
  return (
    <section>
      <SocialAuthButton
        className="social-login-btn social-login-shadow"
        icon={google}
        provider="Google"
      ></SocialAuthButton>
      <SocialAuthButton
        className="social-login-btn social-login-shadow"
        icon={github}
        provider="Github"
      />
    </section>
  );
};
```

And then socialAuthButton can be a lot cleaner and we won't need SocialIcon:

```js
interface SocialButtonProps {
  icon: string;
  provider: string;
  className?: string;
}
const SocialAuthButton = ({ icon, provider, className }: SocialButtonProps) => {
  const onClick = async () => {
    await signIn(provider.toLowerCase());
  };
  return (
    <Button
      onClick={onClick}
      className={twMerge(`social-login-btn social-login-shadow ${className}`)}
    >
      <Image
        src={icon}
        alt={provider}
        width={20}
        height={20}
        className="mr-2"
      />
      Continue with {provider}
    </Button>
  );
};

export default SocialAuthButton;
```

And now that we're passing in just the lowercase name of the provider- we'll directly login instead of getting the default page:

https://github.com/agordon123/git_note/assets/4108484/00c75728-9db9-4a4f-8d92-a022430a6522